databricks/sdk/dbutils.py

```diff
@@ -241,6 +243,75 @@ def __getattr__(self, util) -> '_ProxyUtil':
                           name=util)
 
 
+@dataclass
+class OverrideResult:
+    result: typing.Any
+
+
+def get_local_notebook_path():
+    value = os.getenv("DATABRICKS_SOURCE_FILE")
+    if value is None:
+        raise ValueError(
+            "DABRICKS_SOURCE_FILE environment variable is not set. This is required to get the local notebook path."
+        )
+
+    return value
+
+
+class _OverrideProxyUtil:
+
+    @classmethod
+    def new(cls, path: str):
+        if len(cls.__get_matching_overrides(path)) > 0:
+            return _OverrideProxyUtil(path)
+        return None
+
+    def __init__(self, name: str):
+        self._name = name
+
+    # These are the paths that we want to override and not send to remote dbutils. NOTE, for each of these paths, no prefixes
+    # are sent to remote either. This could lead to unintentional breakage.
+    # Our current proxy implementation (which sends everything to remote dbutils) uses `{util}.{method}(*args, **kwargs)` ONLY.
+    # This means, it is completely safe to override paths starting with `{util}.{attribute}.<other_parts>`, since none of the prefixes
+    # are being proxied to remote dbutils currently.
+    proxy_override_paths = {
+        'notebook.entry_point.getDbutils().notebook().getContext().notebookPath().get()':
+        get_local_notebook_path,
+    }
+
+    @classmethod
+    def __get_matching_overrides(cls, path: str):
+        return list(filter(lambda x: x.startswith(path), list(cls.proxy_override_paths.keys())))
```

Comment:
List comprehension is slightly more readable?
```suggestion
        return [x for x in cls.proxy_override_paths.keys() if x.startswith(path)]
```